internal/donor/donorpage/confirm_your_identity.go

```diff
@@ -22,6 +23,7 @@ func ConfirmYourIdentity(tmpl template.Template, donorStore DonorStore) Handler
 		if r.Method == http.MethodPost {
 			if provided.Tasks.ConfirmYourIdentity.IsNotStarted() {
 				provided.Tasks.ConfirmYourIdentity = task.IdentityStateInProgress
+				provided.IdentityUserData = identity.UserData{}
```

Comment:
I've moved this here from `handleDoNext` as if you don't continue to GOL and return to task list its treated like they've gone down the PO route.